client/repo.c

```diff
@@ -886,65 +902,59 @@ TDNFGetRepoMD(
                               0,
                               NULL);
             BAIL_ON_TDNF_ERROR(dwError);
+            nReplaceRepoMD = 1;
+            if (pszCookie[0])
+            {
+                dwError = SolvCalculateCookieForFile(pszTmpRepoMDFile, pszTmpCookie);
+                BAIL_ON_TDNF_ERROR(dwError);
+                if (!memcmp (pszCookie, pszTmpCookie, sizeof(pszTmpCookie)))
+                {
+                    nReplaceRepoMD = 0;
+                }
+            }
+            nNewRepoMDFile = 1;
+        }
+        if (nNewRepoMDFile)
+        {
+            /* plugin event indicating a repomd download happened */
+            dwError = TDNFEventRepoMDDownloadEnd(
+                          pTdnf,
+                          pRepoData->pszId,
+                          pszRepoMDUrl,
+                          pszTmpRepoMDFile);
+            BAIL_ON_TDNF_ERROR(dwError);
         }
-        /* plugin event indicating a repomd download happened */
-        dwError = TDNFEventRepoMDDownloadEnd(
-                      pTdnf,
-                      pRepoData->pszId,
-                      pszRepoMDUrl,
-                      pszTmpRepoMDFile);
-        BAIL_ON_TDNF_ERROR(dwError);
-
-        nNewRepoMDFile = 1;
 
     }
-    else
+
+    if (metalink && !nReplacebaseURL && !access(pszBaseUrlFile, F_OK))
     {
         /* if metalink url is present, then, we will need to
            set the base url to the url which is used to download the repomd */
-        if(metalink)
-        {
-            dwError = TDNFRepoSetBaseUrl(pTdnf, pRepoData, pszBaseUrlFile);
-            BAIL_ON_TDNF_ERROR(dwError);
-        }
+        dwError = TDNFRepoSetBaseUrl(pTdnf, pRepoData, pszBaseUrlFile);
+        BAIL_ON_TDNF_ERROR(dwError);
     }
 
+    if (nReplaceRepoMD)
+    {
+        /* Remove the old repodata, solvcache and lastRefreshMarker before replacing the new repomd file and metalink files. */
+        TDNFRepoRemoveCache(pTdnf, pRepoData->pszId);
+        TDNFRemoveSolvCache(pTdnf, pRepoData->pszId);
+        TDNFRemoveLastRefreshMarker(pTdnf, pRepoData->pszId);
```

Comment:
Should we remove rpmcache (TDNFRemoveRpmCache()) also based on the flag pTdnf->pConf->nKeepCache?

Reply:
Sure @keerthanakalyan , now we are removing the cached rpms if ```keepcache=true``` is not set in tdnf.conf file. Thanks!